projects/libs/core-library-typescript/test/block-storage/block-storage-sync.spec.ts

```diff
@@ -4,111 +4,99 @@ import * as chaiAsPromised from "chai-as-promised";
 }
 
 describe("Block storage sync", () => {
   let blockStorage: BlockStorage;
   let blockStorageSync: BlockStorageSync;
 
   beforeEach(async () => {
-    try {
-      fsExtra.removeSync(LEVELDB_PATH);
-    } catch (e) { }
-
-    const results = await init();
+    const results  = await init();
     blockStorage = results.blockStorage;
     blockStorageSync = results.blockStorageSync;
   });
 
   afterEach(async () => {
-    await blockStorage.shutdown();
+    if (blockStorage) {
+      await blockStorage.shutdown();
+    }
     blockStorage = undefined;
     blockStorageSync = undefined;
   });
 
   describe("queue", () => {
-    it("accepts new blocks", () => {
+    it("accepts new blocks", async () => {
       blockStorageSync.getQueueSize().should.be.eql(0);
 
-      blockStorageSync.onReceiveBlock(generateBlock(0));
+      const lastBlock = await blockStorage.getLastBlock();
+
+      blockStorageSync.onReceiveBlock(await generateNextNullBlock(lastBlock));
 
       blockStorageSync.getQueueSize().should.be.eql(1);
     });
   });
 
   describe("#appendBlocks", () => {
     it("does not care about order of the blocks", async () => {
-      const blocks = [
-        generateBlock(0),
-        generateBlock(1),
-        generateBlock(2),
-        generateBlock(3)
-      ];
+      const blocks = await generateBlocks(blockStorage, 4);
 
       blockStorageSync.onReceiveBlock(blocks[1]);
       blockStorageSync.onReceiveBlock(blocks[2]);
       blockStorageSync.onReceiveBlock(blocks[0]);
       blockStorageSync.onReceiveBlock(blocks[3]);
 
-      blockStorageSync.getQueueSize().should.be.eql(4);
+      blockStorageSync.getQueueSize().should.eql(4);
       await blockStorageSync.appendBlocks();
-      blockStorageSync.getQueueSize().should.be.eql(0);
+      blockStorageSync.getQueueSize().should.eql(0);
 
-      await blockStorage.getBlocks(0).should.eventually.be.eql(blocks);
+      await blockStorage.getBlocks(0).should.eventually.eql(blocks);
       await blockStorage.getBlocks(4).should.eventually.be.empty;
     });
 
     it("does not return blocks back to the queue in case something happens", async () => {
-      const blocks = [
-        generateBlock(0),
-        generateBlock(1),
-        generateBlock(2),
-        generateBlock(3),
-        generateBlock(5)
-      ];
+      const blocks = await generateBlocks(blockStorage, 6);
+      // removing a block from the array to make it non-continuous
+      blocks.splice(4, 1);
+
 
       blockStorageSync.onReceiveBlock(blocks[1]);
       blockStorageSync.onReceiveBlock(blocks[2]);
       blockStorageSync.onReceiveBlock(blocks[0]);
       blockStorageSync.onReceiveBlock(blocks[4]);
       blockStorageSync.onReceiveBlock(blocks[3]);
 
-      blockStorageSync.getQueueSize().should.be.eql(5);
-
-      const appendBlocksPromise = blockStorageSync.appendBlocks();
-      try {
-        await appendBlocksPromise;
-      } catch (e) {
-      }
+      blockStorageSync.getQueueSize().should.eql(5);
```

Comment:
Let's try to stick to `expect`? There is no benefit of using both at the same time all around the project :(

Reply:
I do, it's just that this module was already using should() and I didn't want to touch it too much